lib/run/summary.js

```diff
@@ -223,14 +233,22 @@ _.assign(RunSummary, {
         emitter.on('request', function (err, o) {
             if (!_.get(o, 'cursor.ref')) { return; }
 
-            var execution = cache[o.cursor.ref] = (cache[o.cursor.ref] || {});
-
-            executions.push(_.assignIn(execution, {
-                cursor: o.cursor,
-                request: o.request,
-                response: o.response,
-                id: _.get(o, 'item.id')
-            }, err && {
+            var execution = cache[o.cursor.ref] = (cache[o.cursor.ref] || {}),
+                executionResponse = {
+                    cursor: o.cursor,
+                    request: o.request,
+                    response: o.response,
```

Comment:
Why don't we add response only when the below `if` condition is false, instead of adding response and then changing it to empty buffer?